web3/contract/utils.py

```diff
@@ -365,6 +373,46 @@ def get_function_by_identifier(
     return fns[0]
 
 
+def find_events_by_identifier(
```

Comment:
nit: Should we rename this to `get_events_by_identifier` since the singular is `get_event_by_identifier`?

Reply:
There's a convention that follows the `find_*` used for lookups which return a list of items, and `get_*` when expecting a single result. The get method takes the resulting list from the find and just returns a single element (if there is not a single result in the list, the get method raises an exception).